satpy/readers/goes_nc.py

```diff
@@ -0,0 +1,918 @@
+                proj_dict=proj_dict,
+                x_size=np.rint((urx - llx) / pix_size).astype(int),
+                y_size=np.rint((ury - lly) / pix_size).astype(int),
+                area_extent=area_extent)
+
+            return area_def
+        else:
+            return None
+
+    @property
+    def start_time(self):
+        """Start timestamp of the dataset"""
+        dt = self.nc['time'].dt
+        return datetime(year=dt.year, month=dt.month, day=dt.day,
+                        hour=dt.hour, minute=dt.minute,
+                        second=dt.second, microsecond=dt.microsecond)
+
+    @property
+    def end_time(self):
+        """End timestamp of the dataset"""
+        if self.sector is not None:
+            return self.start_time + SCAN_DURATION[self.sector]
+
+        return self.start_time
+
+    @property
+    def resolution(self):
+        """Specify the spatial resolution of the dataset.
+
+        Channel 13_3's spatial resolution changes from one platform to another
+        while the wavelength and file format remain the same. In order to
+        avoid multiple YAML reader definitions for the same file format,
+        read the channel's resolution from the file instead of defining it
+        in the YAML dataset. This information will then be used by the YAML
+        reader to complement the YAML definition of the dataset.
+
+        Returns:
+            Spatial resolution in kilometers
+        """
+        return 1000. * self.nc['lineRes'].values
+
+    def get_shape(self, key, info):
+        """Get the shape of the data
+
+        Returns:
+            Number of lines, number of columns
+        """
+        return self.nlines, self.ncols
+
+    @property
+    def meta(self):
+        """Derive metadata from the coordinates"""
+        # Use buffered data if available
+        if self._meta is None:
+            lat = self.nc['lat']
+            earth_mask = self._get_earth_mask(lat)
+            crow, ccol = self._get_nadir_pixel(earth_mask=earth_mask,
+                                               sector=self.sector)
+            lat0 = lat.values[crow, ccol] if crow is not None else None
+            yaw_flip = self._is_yaw_flip(lat)
+            del lat
+
+            lon = self.nc['lon']
+            lon0 = lon.values[crow, ccol] if crow is not None else None
+            area_def_uni = self._get_area_def_uniform_sampling(
+                lon0=lon0, channel=self.gvar_channel)
+            del lon
+
+            self._meta = {'earth_mask': earth_mask,
+                          'yaw_flip': yaw_flip,
+                          'lat0': lat0,
+                          'lon0': lon0,
+                          'nadir_row': crow,
+                          'nadir_col': ccol,
+                          'area_def_uni': area_def_uni}
+        return self._meta
+
+    def get_dataset(self, key, info, out=None, xslice=slice(None),
+                    yslice=slice(None)):
```

Comment:
`xslice` and `yslice` are actually deprecated and are never passed. If you want you could clean up the logic in here so it doesn't use them. Especially since they should be used in the opposite order, so `my_arr[yslice, xslice]`.

Reply:
`out` is also deprecated. You can just return the DataArray at the end of the method.

Reply:
Oh, good catch! I'll remove it.